packages/cdktf-cli/lib/cdktf-project.ts

```diff
@@ -325,7 +464,43 @@ export class CdktfProject {
     return stack.currentPlan;
   }
 
+  private async execute(
+    method: "deploy" | "destroy",
+    next: () => Promise<CdktfStack | undefined>,
+    parallelism: number
+  ) {
+    const maxParallelRuns = parallelism === -1 ? Infinity : parallelism;
+    while (this.stacksToRun.filter((stack) => stack.isPending).length > 0) {
+      const runningStacks = this.stacksToRun.filter((stack) => stack.isRunning);
+      if (runningStacks.length >= maxParallelRuns) {
+        await Promise.race(runningStacks.map((s) => s.currentWorkPromise));
+      } else {
+        const nextRunningExecutor = await next();
+        if (!nextRunningExecutor) {
+          // In this case we have no pending stacks, but we also can not find a new executor
+          break;
+        }
+
+        method === "deploy"
+          ? nextRunningExecutor.deploy()
+          : nextRunningExecutor.destroy();
+      }
+    }
+
+    // We end the loop when all stacks are started, not we need to wait for them to be done
```

Comment:
what? I don't understand the second part of the sentence 😅

Reply:
typo, not should be now